scripts/scil_project_streamlines_to_map.py

```diff
@@ -0,0 +1,232 @@
+"""
+
+import argparse
+import logging
+import os
+
+import nibabel as nib
+from nibabel.streamlines import ArraySequence
+import numpy as np
+
+from scilpy.io.image import assert_same_resolution
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.utils import (add_overwrite_arg,
+                             assert_inputs_exist,
+                             assert_output_dirs_exist_and_empty,
+                             add_reference_arg,
+                             load_matrix_in_any_format)
+from scilpy.utils.filenames import split_name_with_nii
+from scilpy.tractanalysis.streamlines_metrics import \
+    compute_tract_counts_map
+from scilpy.tractanalysis.uncompress import uncompress
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_bundle',
+                   help='Fiber bundle file.')
+    p.add_argument('out_folder',
+                   help='Folder where to save endpoints metric.')
+
+    p1 = p.add_mutually_exclusive_group(required=True)
+    p1.add_argument('--in_metrics', nargs='+', default=[],
+                    help='Nifti metric(s) to compute statistics on.')
+    p1.add_argument('--use_dps', metavar='DPS_KEY', nargs='+',
+                    help='Use the data_per_streamline (scalar) from file, '
+                         'e.g. commit_weights.')
+    p1.add_argument('--use_dpp', metavar='DPP_KEY', nargs='+', default=[],
+                    help='Use the data_per_point (scalar) from file.')
+    p1.add_argument('--load_dps', metavar='DPS_KEY', nargs='+', default=[],
+                    help='Load data per streamline (scalar) .txt or .npy.')
+    p1.add_argument('--load_dpp', metavar='DPP_KEY', nargs='+', default=[],
+                    help='Load data per point (scalar) from .txt or .npy.')
+
+    p.add_argument('--from_wm', action='store_true',
+                   help='Project metrics from whole streamlines coverage.')
+    p.add_argument('--to_wm', action='store_true',
+                   help='Project metrics into streamlines coverage.')
+    add_reference_arg(p)
+    add_overwrite_arg(p)
+    return p
+
+
+def _compute_streamline_mean(cur_ind, cur_min, cur_max, data):
+    # From the precomputed indices, compute the binary map
+    # and use it to weight the metric data for this specific streamline.
+    cur_range = tuple(cur_max - cur_min)
+    streamline_density = compute_tract_counts_map(ArraySequence([cur_ind]),
+                                                  cur_range)
+    streamline_data = data[cur_min[0]:cur_max[0],
+                           cur_min[1]:cur_max[1],
+                           cur_min[2]:cur_max[2]]
+    streamline_average = np.average(streamline_data,
+                                    weights=streamline_density)
+    return streamline_average
+
+
+def _process_streamlines(streamlines, just_endpoints):
```

Comment:
This probably has a place in the library and/or use functions that are already in the library